ktor-retrofit/src/main/kotlin/com/bnorm/ktor/retrofit/RetrofitService.kt

```diff
@@ -170,6 +186,23 @@ private fun Route.process(service: Any, function: KFunction<*>) {
   TODO("implement the rest of the function annotations : $annotations")
 }
 
+private fun Route.checkRegisteredRoute(
+  method: HttpMethod,
+  baseUrl: Route,
+  path: String,
+  serviceInterface: KClass<*>,
+  methodName: String
+) {
+  val registeredRoutes = attributes[injectorKey]
+  val newRoute = RegisteredRoute(method, baseUrl.toString(), path, serviceInterface.qualifiedName, methodName)
+  val registeredPaths = registeredRoutes.getOrPut(method) { mutableSetOf() }
+  val registeredPath = registeredPaths.firstOrNull { registeredPath -> registeredPath == newRoute }
+  if (registeredPath != null) {
+    throw IllegalStateException("@${method.value}(\"$path\") is already registered in ${registeredPath.serviceInterface}::${registeredPath.methodName}")
+  }
+  registeredRoutes[method] = registeredPaths.apply { add(newRoute) }
```

Comment:
The add is required but the put into the map is not required as it is handled by the `getOrPut` call.

Reply:
We need to add the new route though, the `getOrPut` only puts in the empty set if there is no value yet. This second part adds it to the map 

Unless I'm missing something